game-core/src/main/java/games/strategy/engine/framework/message/PlayerListing.java

```diff
@@ -98,7 +106,10 @@ public String getGameRound() {
     return m_gameRound;
   }
 
-  public Map<String, String> getLocalPlayerTypes() {
-    return m_localPlayerTypes;
+  public Map<String, PlayerType> getLocalPlayerTypeMap() {
+    return m_localPlayerTypes.entrySet()
+        .stream()
+        .collect(Collectors.toMap(Entry::getKey, e -> PlayerType.fromLabel(e.getValue())));
```

Comment:
The reason the objects are converted back and forth is compatibility right?

Reply:
Yes, but note we only need to keep compatibility at a class variable level. methods we are free to change in this case.